assets/js/src/settings/pages/key_activation/index.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useContext } from 'react';
+import MailPoet from 'mailpoet';
+import { useSelector, useAction, useSetting } from 'settings/store/hooks';
+import { GlobalContext } from 'context';
+import { t } from 'common/functions';
+import { MssStatus } from 'settings/store/types';
+import { Label, Inputs } from 'settings/components';
+import SetFromAddressModal from 'common/set_from_address_modal';
+import { KeyMessages, MssMessages, PremiumMessages } from './messages';
+
+export default function KeyActivation() {
+  const { notices } = useContext<any>(GlobalContext);
+  const state = useSelector('getKeyActivationState')();
+  const setState = useAction('updateKeyActivationState');
+  const verifyMssKey = useAction('verifyMssKey');
+  const verifyPremiumKey = useAction('verifyPremiumKey');
+  const installPremiumPlugin = useAction('installPremiumPlugin');
+  const activatePremiumPlugin = useAction('activatePremiumPlugin');
+  const [senderAddress] = useSetting('sender', 'address');
+  const [unauthorizedAddresses] = useSetting('authorized_emails_addresses_check');
+
+  const verifyKey = async (event) => {
+    if (!state.key) {
+      notices.error(<p>{t('premiumTabNoKeyNotice')}</p>, { scroll: true });
+      return;
+    }
+    const isTrusted = event.isTrusted;
+    await setState({
+      mssStatus: null,
+      premiumStatus: null,
+      premiumInstallationStatus: null,
+    });
+    MailPoet.Modal.loading(true);
+    const mssStatus: MssStatus = (await verifyMssKey(state.key, isTrusted)) as any;
+    await verifyPremiumKey(state.key);
+    MailPoet.Modal.loading(false);
+    if (isTrusted) {
+      const authorizedAddressNeeded = !senderAddress || unauthorizedAddresses;
+      if (mssStatus === 'valid_mss_active' && authorizedAddressNeeded) {
```

Comment:
One more thing - I noticed that when you have an unauthorized email, then click "Verify", then in the modal add an authorized email and then click "Verify" again, it will display the modal again. But it shouldn't. Also, I think once I had to click twice to actually display the modal (not sure how to replicate now).

Is this caused by using `senderAddress` and `unauthorizedAddresses` which could have changed for the next render cycle but not within the run here in the callback itself? I think that may be the issue - a "wrong" usage of "state" vars.

Reply:
This happens because the modal is shown based on data on the store, but when the new authorized address is sent to the API, the store is not updated, so it shows the modal again. I fixed that by passing a `setAuthorizedAddress` callback to the modal so that it can save the new address on the store.